From 837c7f3d9f7715d308941fcf2ec797231b9f84af Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:48 -0400
Subject: [usb] usb3 ports do not support FEAT_C_ENABLE

Message-id: <1410887463-20674-517-git-send-email-dzickus@redhat.com>
Patchwork-id: 95348
O-Subject: [RHEL7 PATCH 516/591] usb: usb3 ports do not support FEAT_C_ENABLE
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=821903

commit 6908058469e3253637894d1cb3e2581870c77e1d
Author: Dan Williams <dan.j.williams@intel.com>
Date:   Tue May 20 18:09:10 2014 -0700

    usb: usb3 ports do not support FEAT_C_ENABLE

    The port pm_runtime implementation unconditionally clears FEAT_C_ENABLE
    after clearing PORT_POWER, but the bit is reserved on usb3 hub ports.
    We expect khubd to be prevented from running because the port state is
    not RPM_ACTIVE, so we need to clear any errors for usb2 ports.

    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/port.c b/drivers/usb/core/port.c
index f41f051..fb83c2c 100644
--- a/drivers/usb/core/port.c
+++ b/drivers/usb/core/port.c
@@ -142,7 +142,8 @@ static int usb_port_runtime_suspend(struct device *dev)
  set_bit(port1, hub->busy_bits);
  retval = usb_hub_set_port_power(hdev, hub, port1, false);
  usb_clear_port_feature(hdev, port1, USB_PORT_FEAT_C_CONNECTION);
- usb_clear_port_feature(hdev, port1, USB_PORT_FEAT_C_ENABLE);
+ if (!port_dev->is_superspeed)
+  usb_clear_port_feature(hdev, port1, USB_PORT_FEAT_C_ENABLE);
  clear_bit(port1, hub->busy_bits);
  usb_autopm_put_interface(intf);
 
-- 
1.7.1